diff --git a/dist/index.js b/dist/index.js
index 8790fb45aac8d5a793842015b3b3c9c17e588d37..d9c07cb87cc72b1ca409cc2a2798856e8b2ad377 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -159,6 +159,7 @@ function createCli({ router, ...params }) {
         const configureCommand = (command, procedurePath, { meta, parsedProcedure, incompatiblePairs, procedure }) => {
             const optionJsonSchemaProperties = (0, json_schema_1.flattenedProperties)(parsedProcedure.optionsJsonSchema);
             command.exitOverride(ec => {
+                if (ec.code !== 'commander.helpDisplayed') console.log(command.helpInformation());
                 _process.exit(ec.exitCode);
                 throw new errors_1.FailedToExitError(`Command ${command.name()} exitOverride`, { exitCode: ec.exitCode, cause: ec });
             });
@@ -166,11 +167,8 @@ function createCli({ router, ...params }) {
                 writeOut: str => {
                     logger.info?.(str);
                 },
-                writeErr: str => {
-                    logger.error?.(str);
-                },
+                writeErr: () => {},
             });
-            command.showHelpAfterError();
             if (meta.usage)
                 command.usage([meta.usage].flat().join('\n'));
             if (meta.examples)
